site/_wiki/Documentation/ConfigurationGuide.md

```diff
@@ -0,0 +1,418 @@
+---
+title: "Guide: Adding support for a new tablet"
+---
+
+## Prelude
+
+This guide will step you through how to make a configuration for a tablet. Some
+tablets can take a few minutes to set up, while others can require analyzing
+the USB packets sent by the manufacturers tablet driver among other things.
+
+It is probably worthwhile to keep the [Tablet configuration reference] open in
+another tab. This guide will assume that you are aware of the details provided
+on that page.
+
+[Tablet configuration reference]: {% link _wiki/Development/Configurations.md %}
+
+## Preparing your system for adding support
+
+It is highly recommended to remove (or unload) your existing tablet drivers
+before proceeding.
+
+### Uninstalling drivers on Windows
+
+Tablet drivers are often automatically installed by Windows.
+
+Running [TabletDriverCleanup] is the easiest way to ensure that
+OpenTabletDriver has access to the tablet.
+
+[TabletDriverCleanup]: {{ site.data.links.external.TabletDriverCleanup.latest }}
+
+### Unloading drivers on Linux
+
+If you've installed OpenTabletDriver from an official package, the stock tablet
+drivers are already blacklisted.
+
+Still, run the following command to ensure that the upstream drivers are unloaded:
+
+```sh
+sudo rmmod hid_uclogic wacom 2>/dev/null
+```
+
+You likely need to set up udev rules so that your user can access the raw
+device directly.
+
+Normally, this is autogenerated by the build process and uses the officially
+supported configs.
+
+You can define these manually for now, and put them in a local udev rule file,
+e.g. `/etc/udev/rules.d/91-opentabletdriver-adds.rules`:
+
+```conf
+KERNEL=="hidraw*", ATTRS{idVendor}=="b33f", ATTRS{idProduct}=="c4k3", TAG+="uaccess", TAG+="udev-acl"
+SUBSYSTEM=="usb", ATTRS{idVendor}=="b33f", ATTRS{idProduct}=="c4k3", TAG+="uaccess", TAG+="udev-acl"
+```
+
+Make sure to change the values of `idVendor` and `idProduct` as appropriate.
+You can get them with `lsusb`.
+
+> Note: `idVendor` and `idProduct` must be in base-16 (hexadecimal) as opposed
+  to the base-10 format used in OpenTabletDriver configurations.
+
+See `man 7 udev` for systemd udev syntax. If you're not using systemd udev, see
+your distro documentation, or change the appropriate usb and hidraw permissions
+manually.
+
+### MacOS
+
+TODO, probably just uninstall existing drivers?
+
+## Configuration Template
+
+Getting a configuration template is usually simple if OpenTabletDriver supports
+other tablets from your manufacturer, especially if it's in the same line of
+models.
+
+- Copy a "neighbouring" configuration from the
+  `OpenTabletDriver.Configurations/Configurations/` folder
+- Rename the file to be `<model name>.json`, e.g. `H420.json`
+
+If your manufacturer does not have an existing template, please use the
+[Configuration skeleton] or copy any other tablet configuration and follow the
+guide carefully.
+
+> Note that the configuration format between OpenTabletDriver 0.6 (`0.6.x`) and
```

Comment:
You don't need plugins to verify functionality, you can do that all in the tablet debugger, however there are more reasons to steer away from master than that, one notable reason being the lack of the device string reader.

Reply:
Fair enough, I think for me perhaps being the first time I was doing a config I got a bit nervous that the reason pressure sensitivity wasn't working outside the Tablet Debugger was that I had missed some firmware or HID reference since the tablet config I based mine on had 2 different firmware references but I only had 1.

Reply:
Plugins are definitely a bit out of scope. If there's anything that can be improved in terms of setting up new configurations it should be fixed directly in the driver if possible.

Reply:
Yep agreed, I was more just mentioning since that was the headspace I was in without any prior knowledge of how to set one up. In hindsight it seems obvious now that if you can see it working in the debugger it's fine